docroot/WEB-INF/src/com/liferay/course/service/impl/CourseServiceImpl.java

```diff
@@ -49,51 +56,109 @@
+			course.setName(name);
+			course.setDescription(description);
+			course.setLecturer(lecturer);
+			course.setDuration(duration);
+			course.setStatus(status);
+			return CourseLocalServiceUtil.addCourse(course);
+		}
+		else {
+
+			return null;
+		}
 
-		CourseLocalServiceUtil.addCourse(course);
 	}
 
-	public void deleteCourse(long id) throws Exception {
+	@Override
+	public Course deleteCourse(long groupId, long id) throws Exception {
+
+		if (getPermissionChecker().hasPermission(groupId, "com.liferay.course.model.Course", groupId, "DELETE_COURSE")) {
+
+			return CourseLocalServiceUtil.deleteCourse(id);
+		}
+		else {
+
+			return null;
+		}
 
-		CourseLocalServiceUtil.deleteCourse(id);
 	}
 
+	@Override
 	public List<Course> getAllCourses() throws Exception {
 
 		return CourseLocalServiceUtil.getCourses(-1, -1);
 	}
 
+	@Override
 	public Course getCourseById(long id) throws Exception {
 
 		return CourseLocalServiceUtil.getCourse(id);
 	}
 
+	@Override
+	public List<Course> getCoursesByStatus(int status) throws SystemException {
+
+		return CourseLocalServiceUtil.getCoursesByStatus(status);
+	}
+
+	@Override
 	public int getCoursesCount() throws Exception {
 
 		return CourseLocalServiceUtil.getCoursesCount();
 	}
 
-	public void updateCourse(long id, String name, String description, String lecturer, int duration, boolean status)
-			throws Exception {
+	@Override
+	public Course updateCourse(long groupId, long id, String name, String description, String lecturer, int duration,
+		int status) throws Exception {
+
+		validator(name, description, lecturer, duration, status);
+
+		if (getPermissionChecker().hasPermission(groupId, "com.liferay.course.model.Course", groupId, "UPDATE_COURSE")) {
+
+			Course course = CourseLocalServiceUtil.fetchCourse(id);
+			course.setName(name);
+			course.setDescription(description);
+			course.setLecturer(lecturer);
+			course.setDuration(duration);
+			course.setStatus(status);
+			return CourseLocalServiceUtil.updateCourse(course);
+		}
+		else {
+
+			return null;
+		}
+	}
+
+	protected void validator(String name, String description, String lecturer, int duration, int status)
+		throws PortalException {
+
+		if (Validator.isNull(name) || name.length() > 75) {
+			throw new EntryNameException("");
+		}
+
+		if (description.length() > 2000) {
+			throw new EntryDescriptionException();
+		}
 
-		Course course = CourseLocalServiceUtil.fetchCourse(id);
+		if (Validator.isNull(lecturer) || lecturer.length() > 75) {
+			throw new EntryLecturerException();
+		}
 
-		course.setName(name);
-		course.setDescription(description);
-		course.setLecturer(lecturer);
-		course.setDuration(duration);
-		course.setStatus(status);
+		if (duration < 1 || duration > 40) {
+			throw new EntryDurationException();
+		}
 
-		CourseLocalServiceUtil.updateCourse(course);
+		if (Validator.isNull(status)) {
+			throw new EntryDurationException();
+		}
 	}
 }
```

Comment:
Please follow the guide:
checkPermission() // Throws if fail
Do the Business...